# This file is part of HiGFXback

# requires
REQUIRES="lxml-build make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# build
make -C api

# install
install -d $DESTDIR/dfb/include/EGL
install -m 644 api/EGL/egl.h $DESTDIR/dfb/include/EGL
install -m 644 api/EGL/eglext.h $DESTDIR/dfb/include/EGL
install -m 644 api/EGL/eglplatform.h $DESTDIR/dfb/include/EGL
install -d $DESTDIR/dfb/include/KHR
install -m 644 api/KHR/khrplatform.h $DESTDIR/dfb/include/KHR
install -d $DESTDIR/dfb/share/egl/registry
install -m 644 api/egl.xml $DESTDIR/dfb/share/egl/registry

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/egl-registry-build.pc << EOF
Name: EGL-Registry
Version: 20180609
Description: EGL API Registry
Requires: $REQUIRES

devel=\\
/dfb/include/EGL/egl.h \\
/dfb/include/EGL/eglext.h \\
/dfb/include/EGL/eglplatform.h \\
/dfb/include/KHR/khrplatform.h \\
/dfb/share/egl/registry/egl.xml
EOF

exit
--- EGL-Registry-20180609.orig/api/EGL/eglplatform.h
+++ EGL-Registry-20180609/api/EGL/eglplatform.h
@@ -67,7 +67,13 @@
  * implementations.
  */
 
-#if defined(_WIN32) || defined(__VC32__) && !defined(__CYGWIN__) && !defined(__SCITECH_SNAP__) /* Win32 and WinCE */
+#if defined(EGL_NO_PLATFORM_SPECIFIC_TYPES)
+
+typedef void *EGLNativeDisplayType;
+typedef void *EGLNativePixmapType;
+typedef void *EGLNativeWindowType;
+
+#elif defined(_WIN32) || defined(__VC32__) && !defined(__CYGWIN__) && !defined(__SCITECH_SNAP__) /* Win32 and WinCE */
 #ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN 1
 #endif
@@ -110,6 +116,12 @@
 typedef intptr_t EGLNativeWindowType;
 typedef intptr_t EGLNativePixmapType;
 
+#elif defined(__unix__) && defined(EGL_NO_X11)
+
+typedef void             *EGLNativeDisplayType;
+typedef khronos_uintptr_t EGLNativePixmapType;
+typedef khronos_uintptr_t EGLNativeWindowType;
+
 #elif defined(__unix__) || defined(USE_X11)
 
 /* X11 (tentative)  */
--- EGL-Registry-20180609.orig/api/Makefile
+++ EGL-Registry-20180609/api/Makefile
@@ -16,7 +16,7 @@
 EGLHEADERS = EGL/egl.h EGL/eglext.h
 
 # Generation tools
-PYTHON = python
+PYTHON = python3
 PYFILES = genheaders.py reg.py
 REGISTRY = egl.xml
 GENOPTS =
--- EGL-Registry-20180609.orig/api/egl.xml
+++ EGL-Registry-20180609/api/egl.xml
@@ -554,7 +554,9 @@
         <enum value="0x31D7" name="EGL_PLATFORM_GBM_MESA" alias="EGL_PLATFORM_GBM_KHR"/>
         <enum value="0x31D8" name="EGL_PLATFORM_WAYLAND_KHR"/>
         <enum value="0x31D8" name="EGL_PLATFORM_WAYLAND_EXT" alias="EGL_PLATFORM_WAYLAND_KHR"/>
-            <unused start="0x31D9" end="0x31DC"/>
+            <unused start="0x31D9" end="0x31DA"/>
+        <enum value="0x31DB" name="EGL_PLATFORM_DIRECTFB_EXT"/>
+            <unused start="0x31DC" end="0x31DC"/>
         <enum value="0x31DD" name="EGL_PLATFORM_SURFACELESS_MESA"/>
             <unused start="0x31DE" end="0x31DF"/>
     </enums>
@@ -2365,6 +2367,11 @@
                 <enum name="EGL_PLATFORM_X11_SCREEN_EXT"/>
             </require>
         </extension>
+        <extension name="EGL_EXT_platform_directfb" supported="egl">
+            <require>
+                <enum name="EGL_PLATFORM_DIRECTFB_EXT"/>
+            </require>
+        </extension>
         <extension name="EGL_EXT_protected_content" supported="egl">
             <require>
                 <enum name="EGL_PROTECTED_CONTENT_EXT"/>
--- EGL-Registry-20180609.orig/api/genheaders.py
+++ EGL-Registry-20180609/api/genheaders.py
@@ -75,10 +75,10 @@
 startTime = None
 def startTimer():
     global startTime
-    startTime = time.clock()
+    startTime = time.process_time()
 def endTimer(msg):
     global startTime
-    endTime = time.clock()
+    endTime = time.process_time()
     if (timeit):
         write(msg, endTime - startTime)
         startTime = None
@@ -156,8 +156,8 @@
     '** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.',
     '*/',
     '/*',
-    '** This header is generated from the Khronos OpenGL / OpenGL ES XML',
-    '** API Registry. The current version of the Registry, generator scripts',
+    '** This header is generated from the Khronos EGL XML API Registry.',
+    '** The current version of the Registry, generator scripts',
     '** used to make the header, and the header can be found at',
     '**   http://www.khronos.org/registry/egl',
     '**',
